dev.py

```diff
@@ -16,63 +16,115 @@
 This is different from a normal boostrapping process because the
 buildout egg itself is installed as a develop egg.
 """
+import sys
 
-import os, shutil, sys, subprocess
+if sys.version_info < (2, 7):
+    raise SystemError("Outside Python 2.7, no support for Python 2.x.")
+
+if sys.version_info > (3, ) and sys.version_info < (3, 5):
+    raise SystemError("No support for Python 3.x under 3.5.")
+
+
+import os, shutil, sys, subprocess, tempfile
 
 for d in 'eggs', 'develop-eggs', 'bin', 'parts':
     if not os.path.exists(d):
         os.mkdir(d)
 
+bin_buildout = os.path.join('bin', 'buildout')
+if os.path.isfile(bin_buildout):
+    os.remove(bin_buildout)
+
 if os.path.isdir('build'):
     shutil.rmtree('build')
 
 ######################################################################
 # Make sure we have a relatively clean environment
-try:
-    import pkg_resources, setuptools
-except ImportError:
-    pass
-else:
-    raise SystemError(
-        "Buildout development with a pre-installed setuptools or "
-        "distribute is not supported.")
 
-######################################################################
-# Install distribute
-ez = {}
+if '--no-clean' not in sys.argv:
+    try:
+        import pip
+        print('Remove pip, setuptools, wheel')
+        print('')
+        if subprocess.call(
+            [sys.executable] + ['-m', 'pip', 'uninstall', '-y', 'setuptools', 'pip',  'wheel']
+        ):
+            raise SystemError(
+                "Buildout development with pre-installed pip and setuptools"
+                "Could not uninstall with pip"
+                )
+        return_code = subprocess.call(
+            [sys.executable] + sys.argv + ['--no-clean']
+        )
+        sys.exit(return_code)
+    except ImportError:
+        pass
 
-try:
-    from urllib.request import urlopen
-except ImportError:
-    from urllib2 import urlopen
+    try:
+        import pkg_resources, setuptools, pip
+    except ImportError:
+        pass
+    else:
+        raise SystemError(
+            "Buildout development should not come with pre-installed setuptools or pip"
+            )
 
-# XXX use a more permanent ez_setup.py URL when available.
-exec(urlopen('https://bootstrap.pypa.io/ez_setup.py').read(), ez)
-ez['use_setuptools'](to_dir='eggs', download_delay=0)
+#######################################################################
+def install_pip():
+    print('')
+    print('Install pip')
+    print('')
+    try:
+        from urllib.request import urlopen
+    except ImportError:
+        from urllib2 import urlopen
 
-import pkg_resources, setuptools
-setuptools_path = os.path.dirname(os.path.dirname(setuptools.__file__))
+    tmp = tempfile.mkdtemp()
```

Comment:
```suggestion
    tmp = tempfile.mkdtemp(prefix='buildout-dev-')
```

Reply:
I am fine with this change but do not know why it is better to specify the prefix.
I am eager to learn why.

Reply:
I have scars from when I used to maintain a buildbot for the Zope 3.4 KGS.  My /tmp would fill up with a bunch of random temporary directories from tests that forgot to clean up after themselves, and I had a hard time tracking which package's test suite was responsible.

Asking `mkdtemp()` to tag the created temporary directory with a recognizable prefix (or suffix) is basic courtesy in my book.